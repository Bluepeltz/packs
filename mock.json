{
  "id": "bluearchive",
  "title": "Blue Archive",
  "media": {
    "new": [
      {
        "format": "VIDEO_GAME",
        "popularity": 1000000,
        "externalLinks": [],
        "description": "Blue Archive is a role-playing game developed by Nexon Games, a subsidiary of Nexon. It was released in 2021 for Android and iOS, first in Japan by Yostar and worldwide later that year by Nexon. The game is free-to-play with gacha game mechanics as a means to obtain new characters.\n\nThe player and protagonist is a teacher who was summoned to the academy city of Kivotos by the president of the General Student Council, an extrajudicial committee governing the schools. Following her sudden disappearance, criminal activity rises around Kivotos, and the player is tasked by the remaining members of the council to resolve the issues and help search for the president.\n\n(Source: Wikipedia)",
        "characters": [
          {
            "role": "MAIN",
            "characterId": "hy_m"
          },
          {
            "role": "MAIN",
            "characterId": "idla"
          },
          {
            "role": "MAIN",
            "characterId": "yz-m"
          },
          {
            "role": "SUPPORTING",
            "characterId": "sqhb"
          },
          {
            "role": "SUPPORTING",
            "characterId": "xv9p"
          },
          {
            "role": "SUPPORTING",
            "characterId": "wknt"
          },
          {
            "role": "SUPPORTING",
            "characterId": "w2jf"
          },
          {
            "role": "SUPPORTING",
            "characterId": "i2xa"
          },
          {
            "role": "SUPPORTING",
            "characterId": "c8sg"
          },
          {
            "role": "SUPPORTING",
            "characterId": "oqu7"
          },
          {
            "role": "SUPPORTING",
            "characterId": "lv7c"
          },
          {
            "role": "SUPPORTING",
            "characterId": "ahzd"
          },
          {
            "role": "SUPPORTING",
            "characterId": "gy40"
          },
          {
            "role": "SUPPORTING",
            "characterId": "8ppg"
          },
          {
            "role": "SUPPORTING",
            "characterId": "enhq"
          },
          {
            "role": "SUPPORTING",
            "characterId": "blkq"
          },
          {
            "role": "SUPPORTING",
            "characterId": "i8ku"
          },
          {
            "role": "SUPPORTING",
            "characterId": "2hdl"
          },
          {
            "role": "SUPPORTING",
            "characterId": "inhn"
          },
          {
            "role": "SUPPORTING",
            "characterId": "osup"
          },
          {
            "role": "SUPPORTING",
            "characterId": "fkle"
          },
          {
            "role": "SUPPORTING",
            "characterId": "tzf4"
          },
          {
            "role": "SUPPORTING",
            "characterId": "dmff"
          },
          {
            "role": "SUPPORTING",
            "characterId": "yhxu"
          },
          {
            "role": "SUPPORTING",
            "characterId": "r2b5"
          },
          {
            "role": "SUPPORTING",
            "characterId": "g4mf"
          },
          {
            "role": "SUPPORTING",
            "characterId": "3eyw"
          }
        ],
        "id": "xqzn",
        "relations": [],
        "title": {
          "english": "Blue Archive",
          "alternative": ["BA"]
        },
        "type": "OTHER",
        "images": [
          {
            "url": "https://f005.backblazeb2.com/file/fable-packs/22TUW0Cu"
          }
        ]
      }
    ],
    "conflicts": []
  },
  "characters": {
    "new": [
      {
        "externalLinks": [],
        "name": {
          "english": "Takanashi Hoshino",
          "alternative": ["hoshino"]
        },
        "description": "Takanashi Hoshino is the former vice president and the only third-year of Abydos High School, and the current chairwoman for the Countermeasures Committee.\n\nSpeaking like a senile aunt, she enjoys playing around and napping more than work. Because of this, she usually gets scolded by the rest of the Committee. But, once the mission starts, she protects her comrades by fighting valiantly on the front lines.",
        "age": "17",
        "id": "hy_m",
        "media": [
          {
            "role": "MAIN",
            "mediaId": "xqzn"
          }
        ],
        "images": [
          {
            "url": "https://f005.backblazeb2.com/file/fable-packs/rtfbupny"
          }
        ],
        "gender": "Female"
      }
    ],
    "conflicts": []
  },
  "maintainers": ["228674702414053386", "839261971185664011"],
  "image": "https://f005.backblazeb2.com/file/fable-packs/dkihlBqc",
  "author": "Fable Community",
  "description": "A pack containing the various characters from the game Blue Archive",
  "conflicts": [
    "anilist:161049",
    "anilist:138416",
    "anilist:160589",
    "anilist:161050",
    "anilist:143177"
  ]
}
