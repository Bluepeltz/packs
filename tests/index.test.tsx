import { renderToString } from 'preact-render-to-string';

import { assertSnapshot } from '$std/testing/snapshot.ts';

import Index from '../routes/index.tsx';

import { stub } from '$std/testing/mock.ts';

Deno.test('<Index/>', async (test) => {
  const mathStub = stub(
    Math,
    'random',
    () => 0,
  );

  try {
    const url = new URL('http://localhost:8080');

    await assertSnapshot(
      test,
      renderToString(<Index params={{}} route={''} url={url} data={{}} />),
    );
  } finally {
    mathStub.restore();
  }
});

Deno.test('<Index/> (Logged In)', async (test) => {
  const mathStub = stub(
    Math,
    'random',
    () => 0,
  );

  try {
    const url = new URL('http://localhost:8080');

    await assertSnapshot(
      test,
      renderToString(
        <Index
          params={{}}
          route={''}
          url={url}
          data={{
            id: 'user_id',
            avatar: 'avatar_hash',
            username: 'user_username',
          }}
        />,
      ),
    );
  } finally {
    mathStub.restore();
  }
});

Deno.test('<Index/> (Logged In) (No Avatar)', async (test) => {
  const mathStub = stub(
    Math,
    'random',
    () => 0,
  );

  try {
    const url = new URL('http://localhost:8080');

    await assertSnapshot(
      test,
      renderToString(
        <Index
          params={{}}
          route={''}
          url={url}
          data={{
            id: 'user_id',
            username: 'user_username',
          }}
        />,
      ),
    );
  } finally {
    mathStub.restore();
  }
});
