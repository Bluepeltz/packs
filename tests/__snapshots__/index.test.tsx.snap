export const snapshot = {};

snapshot[`<Index/> 1`] = `'<img src="/icon.png" class="flcss-logo-"/><form method="post" action="/api/login"><button class="flcss-button-" type="submit">Login with Discord<i class="bx bxl-discord-alt flcss-icon-"></i></button></form>'`;

snapshot[`<Index/> (Logged In) 1`] = `'<img src="/icon.png" class="flcss-logo-"/><p>user_username</p><img width="32" height="32" src="https://cdn.discordapp.com/avatars/user_id/avatar_hash.png"/><form method="post" action="/api/logout"><button type="submit">Log Out</button></form>'`;

snapshot[`<Index/> (Logged In) (No Avatar) 1`] = `'<img src="/icon.png" class="flcss-logo-"/><p>user_username</p><img width="32" height="32" src="https://cdn.discordapp.com/embed/avatars/0.png"/><form method="post" action="/api/logout"><button type="submit">Log Out</button></form>'`;
